### ISSUE TITLE
Wanting to use Ultralytics models in the Rust Burn framework but concerned about open-source licensing.

### ISSUE THREAD

**Issue by monkeycc** [ultralytics/ultralytics#21696]
https://github.com/tracel-ai/models/issues/76#issuecomment-3189205277

> Just a note on proper licensing / attribution since ultralytics has an AGPL-3.0 license. I'm not an expert on these things, so correct me if I am wrong:
> 
> * I think a re-implementation of the architecture in Burn can be under MIT/Apache
> * The pre-trained weights definitely need to be under their AGPL-3.0 license
> 
> Or to get a clear response on both, someone could ask the authors.

https://github.com/tracel-ai/burn

Wanting to use Ultralytics models in the Rust Burn framework but concerned about open-source licensing.

From a development perspective,
models from Ultralytics should be usable across various frameworks
under MIT/Apache licenses.

Frameworks under MIT/Apache licenses can call and utilize various models.
However, when encountering models licensed under AGPL-3.0,
the terms of the model’s license must be respected.

This approach facilitates the use of Ultralytics models
by various frameworks.
Any use of AGPL-3.0-licensed models
must comply with their license terms.

I am not a legal professional,
but I believe AI frameworks and models
should have separate licensing terms.

Could Ultralytics make an official announcement stating:
Frameworks under MIT/Apache licenses may call and use Ultralytics models,
but such frameworks must also explicitly state that if they use Ultralytics models
licensed under AGPL-3.0, compliance with the AGPL-3.0 license is required.

Could this achieve a mutually beneficial solution?

Ultralytics expands its market reach
by enabling various AI frameworks to integrate its models,
while still ensuring AGPL-3.0 compliance for applicable models.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @monkeycc, thank you for opening this discussion and for your interest in Ultralytics 🚀! This is an automated response to help gather details and guide next steps. An Ultralytics engineer will also follow up with you soon.

Please review our [Docs](https://docs.ultralytics.com/) for general guidance, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, which may address common questions.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) so we can investigate efficiently.

If this is a licensing or integration ❓ Question (e.g., using Ultralytics models like YOLO11 in other frameworks such as Rust/Burn), please share:
-Exact models and versions you want to use (e.g., YOLO11, tasks, model sizes)
-How you plan to distribute or deploy (research, SaaS, on-prem, commercial use, etc.)
-Whether you’ll use Ultralytics code, pretrained weights, or re-implementations only
-Links to relevant repositories or code snippets demonstrating your approach

Join the Ultralytics community where it suits you best:
-For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧
-For in-depth discussions, check out [Discourse](https://community.ultralytics.com/)
-Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community

## Upgrade

Before proceeding, ensure you’re on the latest `ultralytics` release with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
-Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
-Google Cloud Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
-Amazon Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
-Docker Image. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by glenn-jocher** [comment]
Thanks for raising this—your summary is broadly correct: you can implement the model in Burn under MIT/Apache, but once you use or distribute any Ultralytics YOLO code or pretrained weights, the larger work must be AGPL-3.0 open-sourced or covered by an Enterprise License; this applies to internal, external, local, and networked use. Our published pages are the official position, so we won’t issue framework-specific announcements; please review the Ultralytics License page and the Contributing guide’s Open-Sourcing Your YOLO Project Under AGPL-3.0 section for details.

**Comment by monkeycc** [comment]
"Can the MIT/Apache-licensed Burn framework utilize Ultralytics models without AGPL-3.0 'contaminating' the entire framework? Specifically when dynamically calling models:

Users employing Burn with other models remain unaffected by AGPL-3.0

Users leveraging Ultralytics' AGPL-3.0 models must comply with that license
This approach enables Burn to coexist with multiple licenses while supporting Ultralytics model integration."

**Comment by glenn-jocher** [comment]
Yes—Burn can stay MIT/Apache as long as the core repo does not include or redistribute Ultralytics code or weights; keep any Ultralytics integration as a separate, optional AGPL-3.0 plugin/repo and don’t bundle our pretrained weights. Users who opt in must comply with AGPL-3.0 or hold an Enterprise License; dynamic loading doesn’t change that. For authoritative terms, see the AGPL 3.0 Software License.
